.github/workflows/pr.yaml

```diff
@@ -10,27 +10,41 @@ jobs:
   static-analysis:
     runs-on: macos-15
     steps:
-      - name: Member Check
-        if: ${{ github.event.pull_request.author_association != 'MEMBER' && github.event.pull_request.author_association != 'OWNER' }}
+      - name: Check Write Permissions
+        uses: octokit/request-action@v2.x
+        id: check_permissions
+        with:
+          route: GET /repos/${{ github.repository }}/collaborators/${{ github.triggering_actor }}/permission
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      - name: Debug Permissions Response
+        run: |
+          echo "Permissions raw response: ${{ steps.check_permissions.outputs.data }}"
+      - name: Validate Write Permissions
         run: |
-          echo "Pull Request not triggered by a MSDK team member.  Someone from the team needs to rerun this workflow AFTER it has been deemed safe."
-          exit 1
+          permission=$(echo "${{ fromJson(steps.check_permissions.outputs.data).permission }}")
+          echo "User ${{ github.actor }} has permissions: $permissions"
```

Comment:
I think the echo statements should use triggering_actor as well

Reply:
ah yes, updated